File: fth-1605-assessment/sync-to-oss.py
#!/usr/bin/env python3
import os
from pathlib import Path

OSS_REGION = os.environ.get("OSS_REGION", "oss-cn-shanghai")
OSS_BUCKET = os.environ.get("OSS_BUCKET", "ceping-air")
OSS_PATH_PREFIX = os.environ.get("OSS_PATH_PREFIX", "fth1605")
OSS_ACCESS_KEY_ID = os.environ.get("OSS_ACCESS_KEY_ID", "")
OSS_ACCESS_KEY_SECRET = os.environ.get("OSS_ACCESS_KEY_SECRET", "")

def sync_fth1605_to_oss():
    try:
        import oss2
        auth = oss2.Auth(OSS_ACCESS_KEY_ID, OSS_ACCESS_KEY_SECRET)
        endpoint = f"https://{OSS_REGION}.aliyuncs.com"
        bucket = oss2.Bucket(auth, endpoint, OSS_BUCKET)

        output_dir = Path("output")
        if output_dir.exists():
            files = [f for f in output_dir.iterdir() if f.is_file() and not f.name.startswith(".")]
            print(f"在 output 目录找到 {len(files)} 个报告/产物文件，准备同步至 OSS {OSS_BUCKET} -> {OSS_PATH_PREFIX}/reports/ ...")
            for f in files:
                object_key = f"{OSS_PATH_PREFIX}/reports/{f.name}"
                bucket.put_object_from_file(object_key, str(f))
                print(f"[✓] FTH1605 报告文件上传成功: {object_key}")
    except Exception as e:
        print(f"[OSS Error] 上传失败: {e}")

if __name__ == "__main__":
    sync_fth1605_to_oss()

File: public/fth1605/sync-to-oss.mjs
import OSS from "ali-oss";
import fs from "node:fs";
import path from "node:path";

const ossConfig = {
  region: process.env.OSS_REGION || "oss-cn-shanghai",
  bucket: process.env.OSS_BUCKET || "ceping-air",
  accessKeyId: process.env.OSS_ACCESS_KEY_ID || "",
  accessKeySecret: process.env.OSS_ACCESS_KEY_SECRET || "",
  prefix: process.env.OSS_PATH_PREFIX || "fth1605"
};

const client = new OSS({
  region: ossConfig.region,
  accessKeyId: ossConfig.accessKeyId,
  accessKeySecret: ossConfig.accessKeySecret,
  bucket: ossConfig.bucket
});

try {
  const outputDir = path.resolve("output");
  if (fs.existsSync(outputDir)) {
    const files = fs.readdirSync(outputDir);
    console.log(`在 output 目录找到 ${files.length} 个文件，开始同步至阿里云 OSS (${ossConfig.bucket} -> ${ossConfig.prefix}/reports/)...`);
    for (const file of files) {
      const filePath = path.join(outputDir, file);
      if (fs.statSync(filePath).isFile() && !file.startsWith(".")) {
        const objectKey = `${ossConfig.prefix}/reports/${file}`;
        await client.put(objectKey, filePath, {
          headers: { "x-oss-object-acl": "public-read" }
        });
        console.log(`[✓] 报告产物 ${file} 上传成功 -> ${objectKey}`);
      }
    }
  } else {
    console.log("未找到 output 目录");
  }
} catch (err) {
  console.error("OSS 上传失败:", err.message);
}
